### ISSUE TITLE
[React-aria] FocusScope with contain enabled breaks useFocusRing behaviour for input radio / checkbox

### ISSUE THREAD

**Issue by dnaploszek** [adobe/react-spectrum#4288]
# 🐛 Bug Report 

I've build a modal that uses `FocusScope contain` to contain the focus inside of it. However when we're using `useFocusRing` on input type `'radio'` or `'checkbox'`, inside of the modal, the inputs have focus state even though I'm using mouse to select them

## 🤔 Expected Behavior
Focus rings should only appear on keyboard events or autofocus.

## 😯 Current Behavior
Here's a quick demo on how I was able to reproduce it:

https://user-images.githubusercontent.com/17772241/228822138-b6111ccf-67c2-41ee-8084-6934cf73d584.mov

https://codesandbox.io/s/unruffled-wind-rlsexw?file=/src/Radio.js

## 💁 Possible Solution
Not sure yet I'm still reviewing the code, but as far as I could find out there are two events triggered on the input when clicking on the input, one changes the input modality to `virtual` which makes `isFocusVisible = true`. "The weird one" is Focus which I presume is somehow triggered within the FocusScope. I don't fully understand why the click is also considered as "virtual" though 🤔 

<img width="630" alt="image" src="https://user-images.githubusercontent.com/17772241/228822709-f4d3f727-60d4-4b8d-8673-82f9363d3241.png">

I'll investigate it a bit more too, but hoping for a hint if I'm just doing something completely wrong. I had a suspicion that it's because I'm wrapping both span and input with <label> but even with label outside it still happens. 

## 🔦 Context
I want to have nice accessible modals with forms inside of them, but I don't want the focus rings appear when I'm using mouse.

## 💻 Code Sample
Linked above but here they are again 😅 : 
https://user-images.githubusercontent.com/17772241/228822138-b6111ccf-67c2-41ee-8084-6934cf73d584.mov

https://codesandbox.io/s/unruffled-wind-rlsexw?file=/src/Radio.js

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-aria  | 3.23.1 |
| Browser          | Chrome 111.0.5563.110 | 
| Operating System | MacOS 12.1 | 

## 🧢 Your Company/Team
Not affiliated with adobe 🤔 

## 🕷 Tracking Issue (optional)
🤷🏻

**Comment by snowystinger** [comment]
I think this is related or a duplicate of https://github.com/adobe/react-spectrum/issues/3675
I noticed that it only happens when clicking the label, not the input itself.

**Comment by dnaploszek** [comment]
Yeah it is certainly connected with the label. But it does not happen when the label + input is outside of the FocusScope. So not sure if it's 100% related to the same issue. Funny thing is that on my example it happens only when you switch between the inputs. 

Our current workaround was to remove the label (and have an interactive div to handle clicks with usePress 🙈 ), keeping all the other accessibility props (labelled by etc.). So I'm keen to hear on a better solution.

**Comment by snowystinger** [comment]
I looked at the codesandbox a little, I think if you use our hooks for RadioGroup/Radio that this issue will go away.
Or even just add a usePress handler https://codesandbox.io/s/funny-pateu-bi7k8f?file=/src/Radio.js
Btw, it will be easier to build components using our hooks, here's a preview of our react-aria-components https://reactspectrum.blob.core.windows.net/reactspectrum/4d4c31ab4d9ecb0eb3201436605bf9b5e43aab5a/docs/react-aria/RadioGroup.html

There's still something to address here, so I'll leave the issue open. But there is a workaround using our hooks.

**Comment by dnaploszek** [comment]
Yeah internally we were using your hooks for the RadioGroups, the demo was just as "bare" as possible to show the issue.

We were also using a very "questionable" approach of wrapping quite complex card "selectable" structures with a `<label>` and relied on the browser default callback for clicking on it to select the input itself. I ended up using just a `<div>` wrapper with `usePress` hook and the issue is no more. So yep, there are workarounds but I still think the default behaviour is a bit wonky, as browsers do not show outline in my demo. 

Thanks for replying - I have to also add that I love `react-aria` and it saves us tons of time with building our design system! Great work on this library 👏🏻

**Comment by snowystinger** [comment]
Going through old issues. Unfortunately codesandbox seems to have discarded the project, so I can't access it anymore to try with out latest code to re-confirm. I'm going to close this issue, if there's still a problem, maybe we can recreate it in stackblitz or our storybook. Thank you!
